packages/upload/test/adding-files.test.js

```diff
@@ -27,76 +27,67 @@ describe('file list', () => {
+      // it should generate a non-trusted click event on the hidden file input,
+      // at the time of writing Chrome and Firefox still open the file dialog.
+      // Use stub calling `preventDefault` to prevent dialog from opening.
+      inputClickSpy = sinon.stub().callsFake((e) => e.preventDefault());
+      input.addEventListener('click', inputClickSpy);
+    });
+
     it('should open file dialog by click', () => {
-      // This test checks if the 'click' event is synchronously dispatched
-      // on the hidden file input when user clicks Add Files button. The
-      // file dialog is actually not getting opened during testing.
-      //
-      // The mock Add Files button click event fired by this test is not
-      // trusted and hence it should generate a non-trusted click event on
-      // the hidden file input. For security reasons non-IE browsers don't
-      // open file dialog when file input click is non-trusted.
-      const clickSpy = sinon.spy();
-      upload.$.fileInput.addEventListener('click', clickSpy);
-
-      upload.$.addFiles.dispatchEvent(new MouseEvent('click'));
-      expect(clickSpy.calledOnce).to.be.true;
+      click(addFiles);
+      expect(inputClickSpy.calledOnce).to.be.true;
     });
 
     it('should open file dialog by touchend', () => {
-      const clickSpy = sinon.spy();
-      upload.$.fileInput.addEventListener('click', clickSpy);
-
-      const e = new CustomEvent('touchend', { cancelable: true });
-      upload.$.addFiles.dispatchEvent(e);
-      expect(clickSpy.calledOnce).to.be.true;
-      expect(e.defaultPrevented).to.be.true;
+      const event = makeSoloTouchEvent('touchend', null, addFiles);
+      expect(inputClickSpy.calledOnce).to.be.true;
+      expect(event.defaultPrevented).to.be.true;
     });
 
     it('should invoke Polymer.Gestures.resetMouseCanceller before open file dialog', () => {
       // NOTE(platosha): With ES modules we can’t put a spy on the actual
       // Polymer.Gestures.resetMouseCanceller. Have to use a separate
       // wrapper method for testing.
       const spy = sinon.spy(upload, '__resetMouseCanceller');
-
-      const e = new CustomEvent('touchend', { cancelable: true });
-      upload.$.addFiles.dispatchEvent(e);
+      touchend(addFiles);
       expect(spy.calledOnce).to.be.true;
     });
 
-    it('should reset input.value before dialog', (done) => {
-      const input = upload.$.fileInput;
+    it('should reset input.value before dialog', () => {
       // We can't simply assign `files` property of input[type="file"].
       // Tweaking __proto__ to make it assignable below.
       input.__proto__ = HTMLElement.prototype;
       delete input.value;
       input.value = 'foo';
 
-      input.addEventListener('click', () => {
-        expect(input.value).to.be.empty;
-        done();
-      });
-
-      upload.$.addFiles.dispatchEvent(new MouseEvent('click'));
+      addFiles.dispatchEvent(new MouseEvent('click'));
```

Comment:
```suggestion
      click(addFiles);
```

Reply:
After I changed the code, the test started to fail in Firefox because of `composed` flag on the click event.
This indicated an actual issue: the click event should not bubble, as it's handled by the component.